examples/chart/teleport-kube-agent/templates/statefulset.yaml

```diff
@@ -0,0 +1,140 @@
+{{- if .Values.storage.enabled }}
```

Comment:
it's unfortunate that 90% of this file is identical to `deployment.yaml`
we will have drift between the contents of these files over time.

let's at least add a comment at the top of both files as a reminder to update the other one.

Reply:
Yea, TBH I originally wrote it as an ifdef's in the one file for just the changes, but that also seemed like a logical mess to support the 10% differences. IDK which way is better, both seem likely to produce issues.

Reply:
Added in 1fa70acd5e5bc3ad7a2422a235fe9ccaff8a9657

Reply:
We could probably actually achieve this with a Helm template/function - basically define the main chart structure in one place and allow optional includes specific to `Deployment` and `StatefulSet`. That way the code only lives in one file and any changes will be represented in both places.

Reply:
I believe this is what we were discussing, and this is the first way I wrote it. Unfortunately there are 3, maybe 4, differences that occur so it creates a bit of a messy template that removes/adds sections based on whether it's a StatefulSet or Deployment.